### PR TITLE
Migrate from LESS to CSS

### LINKED ISSUE #4994: Extensions defining new importing controllers need to import theme.less from OpenRefine
Some extensions define new "importing controllers" (for instance the gdata and database extension). This is the expected extension point when implementing an importing mechanism which loads data from other sources.

As part of this, extensions need to define parsing pages which look like the importer preview we have in OpenRefine.
For the styling of those parsing pages, both the gdata and database extensions rely on OpenRefine-internal `.less` files that they explicitly import on their own side.

This looks like this:

https://github.com/OpenRefine/OpenRefine/blob/5746951ec069549345eb368ea74a77c3fec9a912/extensions/database/module/styles/theme.less#L30

While this technique avoids duplicating files from the core to the extensions, this is far from ideal because this mechanism is not available to extensions developed outside this repository.

Intuitively, it should be possible for any extension to use the standard styling of the parsing page, without having to redefine it itself. Of course, we still want extensions to be able to add some custom CSS to this page, should they need to.

Ideally, it would be great if we could fix this in a backwards-compatible way (meaning that extensions which have been built for the current situation would keep working with the new system), although this is likely to be difficult. And it is probably not so important since we can easily fix the extensions in this repository.

@antoine2711 @WaltonG this is an issue you are likely to run into (or have already run into?) for the SPARQL extension, no?

### PR DESCRIPTION
**Pull request by Abbe98**
This change migrates OpenRefine's usage of LESS to CSS, to:
1. make progress towards dynamic theming (#3017, etc)
2. make it easier for extensions to use variables/styles defined by core (#4994)
3. make it possible to serve CSS files as is so that one can drop dependencies from butterfly and improve load time

Fixes #4994 (possibly other issues as well)

Changes proposed in this pull request:
- Migrate from LESS to CSS
- CSS formatting

### CODE REVIEW

**Review by elebitzero — COMMENTED** [review]
I tried out the changes locally and the UI looked the same except for the entries in the history sidebar which need the amount of padding they had before.

File: main/webapp/modules/core/styles/project/sidebar.css [hunk 1]
```diff
@@ -145,120 +143,116 @@ div#body div#left-panel a#hide-left-panel-button:hover {
...
-@history_undone_background: @chrome_secondary;
-@history_entry_index_width: 30px;
-
 a.history-entry {
   display: block;
   position: relative;
-  padding: @padding_normal;
-  padding-left:  @padding_normal + @history_entry_index_width;
+  padding: var(--padding-normal);
+  padding-left: var(--padding-normal) + 3px;
```
[the remark this answers is not in the document]
Comment (Abbe98) [reply main/webapp/modules/core/styles/project/sidebar.css:150]: Nice catch! I didn't consider expressions when doing search and replace on these.

**Comment by Abbe98** [comment]
@wetneb I'm picking up the discussion you started in OpenRefine/simile-butterfly/pull/81 regarding breaking changes to the extension interface.

My thinking was to await the 3.8 release cycle to give extensions time to migrate. I, however, don't have more than one affected extension myself and I'm very much wanting this breaking change. I'm not sure how far away 4.0 is and thus a little concerned about keeping up with master and being blocked on the next steps(vector icons, color harmonization, variable adoption, etc).

**Comment by wetneb** [comment]
My understanding is that this change, migrating from LESS to CSS in OpenRefine itself, would not prevent extensions from still using LESS until we remove support for LESS in Butterfly - but maybe I am wrong there?

**Comment by Abbe98** [comment]
@wetneb that's correct, the breaking change here would be that one can't depend on `theme.less`/LESS variables anymore.

To me dropping LESS from Butterfly is a bonus so targeting 4.0 with that change is totally okay!

**Comment by wetneb** [comment]
> the breaking change here would be that one can't depend on theme.less/LESS variables anymore.

I haven't looked into it closely but I think currently extensions actually have to include a copy of theme.less on their side as well, in some situations. [It was done for the Commons extension for instance.](https://github.com/OpenRefine/CommonsExtension/tree/master/module/styles) I find that pretty bad but I guess it is good news for this PR since it hopefully lets them rely on the LESS variables on their side as long as the version of Butterfly used supports LESS.

**Comment by Abbe98** [comment]
> I haven't looked into it closely but I think currently extensions actually have to include a copy of theme.less

A quick look over at openrefine.org/extensions seems to indicate that it's the common solution(the only one I found, most doesn't seem to use `theme.less`) and given that `theme.less` isn't in the technical reference maybe we are on the safe side.

**Review by elebitzero — APPROVED** [review]

**Pull request merged**